File: go/06_linkedlist/listinterface.go
package linkedlist

// ListInterface the interface of a list structure.
type ListInterface interface {
	// 在某节点前插入
	InsertAfter(p *SingleListNode, v interface{}) bool
	// 在某节点后插入
	InsertBefore(p *SingleListNode, v interface{}) bool
	// 在头结点插入
	InsertToHead(v interface{}) bool
	// 在尾节点插入
	InsertToTail(v interface{}) bool
	// 通过下标查询节点
	FindByIndex(index uint) *SingleListNode
	// 删除节点元素
	DeleteNode(p *SingleListNode) bool
}

File: java/06_linkedlist/ListInterface.java
package linkedlist;

/**
 * common methods must contain in list.
 */
public interface ListInterface<ListNode> {

    public ListNode findByIndex(int index);

    public void insertToHead(V value);

    public void insertToTail(V value);

    public void insertBefore(ListNode node, V value);

    public void insertAfter(ListNode node, V value);
}
